website/server/libs/user/validation.js

```diff
@@ -15,6 +15,13 @@ export function nameContainsSlur (username) {
   return false;
 }
 
+export function nameContainsNewline (username) {
+  if (username.includes('\n') || username.includes('\\n')) {
```

Comment:
@Amberlisi can you remind me why we're checking both \n and \\n?

Reply:
When changing your username in the web UI, it automatically escapes the backslash N. So it doesn't see it as a newline. Which is why you have to search for \\\n

The API works differently. It doesn't automatically escape it, so you have to check for a newline which is \\n.

Not sure if that makes sense

Reply:
Hmm I see but the moment is escaped (so the api gets `\\n`) is this an issue anymore? since it's escaped it shouldn't result in a visible new line. Not a big deal but I'm not sure the second check is necessary

Reply:
This depends on what we want the end functionality to be.

Do we want the users to be able to set their name to something that includes a \ + n using the UI, like so?
![image](https://user-images.githubusercontent.com/43859793/87343899-2f7d1b00-c513-11ea-81e9-d3c66dc97c43.png)

Or should the ability to do that in the UI be restricted just like it would be in the API? I would think so, for consistency. If something isn't allowed in the API, I don't think we would want to allow it anywhere else. That's just my opinion and we can go whichever way is desired

Reply:
> Or should the ability to do that in the UI be restricted just like it would be in the API? 

The UI under the hood uses the API , what we want to prevent here it simply people having actual line breaks (that then get displayed ) in their display name or username, if they have the literal `\n` we don't care as long as it doesn't create a new line

Reply:
If that's the case then the second check can be left out. I'll make that change in a moment. If you could help me figure out how to perform tests for this that would be great

Reply:
For the tests you should update https://github.com/HabitRPG/habitica/blob/develop/test/api/v3/integration/user/PUT-user.test.js#L55 to add a new case covering a profile.name that includes `\n` and add a check that it fails like it's done for the other invalid profile names. 

Something similar will have to be done for the route verifying the validity of a display name here https://github.com/HabitRPG/habitica/blob/develop/test/api/v4/user/auth/POST-user_verify_display_name.test.js with the difference that in this case each different invalid "case" has a separate test (identified by an `it(...)` block) so you'll have to add a new one for `\n`

Reply:
Okay, I added tests. Let me know if I did that right and if not what I need to change. I also removed the second check from the validation.